### ISSUE TITLE
gd is not working.

### ISSUE THREAD

**Issue by NEX-S** [glepnir/lspsaga.nvim#429]
### Description
gd is not working

**Expected Behavior**
gd can show floating definition windows

**Actual Behavior**
nvim error 
```
E5108: Error executing lua: .../pack/packer/opt/lspsaga.nvim/lua/lspsaga/definition.lua:67: attempt to concatenate local 'short_name' (a nil value)
stack traceback:
        .../pack/packer/opt/lspsaga.nvim/lua/lspsaga/definition.lua:67: in function <.../pack/packer/opt/lspsaga.nvim/lua/lspsaga/definition.lua:10>
```

**Neovim Built in Behavior**
normal

### Details

<details><summary>Reproduce</summary>

1. nvim -u test.vim the min config that only lspsaga
2. `printf("Hello World");`
3. Cursor on `printf` and press `gd`
4. got error
</details>

<details><summary>Environment</summary>

- nvim --version output: 
```
NVIM v0.8.0-dev+608-g778541067
Build type: RelWithDebInfo
LuaJIT 2.1.0-beta3
Compilation: /usr/bin/gcc-11 -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1 -DNVIM_TS_HAS_SET_MATCH_LIMIT -DNVIM_TS_HAS_SET_ALLOCATOR -O2 -g -Og -g -Wall -Wextra -pedantic -Wno-unused-parameter -Wstrict-prototypes -std=gnu99 -Wshadow -Wconversion -Wdouble-promotion -Wmissing-noreturn -Wmissing-format-attribute -Wmissing-prototypes -Wimplicit-fallthrough -Wvla -fstack-protector-strong -fno-common -fdiagnostics-color=always -DINCLUDE_GENERATED_DECLARATIONS -D_GNU_SOURCE -DNVIM_MSGPACK_HAS_FLOAT32 -DNVIM_UNIBI_HAS_VAR_FROM -DMIN_LOG_LEVEL=3 -I/home/runner/work/neovim/neovim/build/cmake.config -I/home/runner/work/neovim/neovim/src -I/home/runner/work/neovim/neovim/.deps/usr/include -I/usr/include -I/home/runner/work/neovim/neovim/build/src/nvim/auto -I/home/runner/work/neovim/neovim/build/include
Compiled by runner@fv-az120-753

Features: +acl +iconv +tui
See ":help feature-compile"

   system vimrc file: "$VIM/sysinit.vim"
  fall-back for $VIM: "/share/nvim"

Run :checkhealth for more info
```
- Operating system: Manjaro Linux amd64
- lspsaga commit: 

</details>

**Comment by NEX-S** [comment]
i've set `gd` to `vim.keymap.set('n', "gd",    require "lspsaga.definition".preview_definition, opt )`

**Comment by NEX-S** [comment]
`gd` is normal in lua, but in `test.c` it seems not work.

**Comment by NEX-S** [comment]
![image](https://user-images.githubusercontent.com/97107165/184496452-f6327ccc-bae8-4ce1-a23d-237186d40791.png)
if I change `#include <stdio.h>` to the function declaration in stdio.h, gd works fine, i guess is gd can't find the `printf` in other files

**Comment by glepnir** [comment]
version 2 had fixed .

**Comment by NEX-S** [comment]
Seems it's still not working...
```
Error executing vim.schedule lua callback: .../pack/packer/opt/lspsaga.nvim/lua/lspsaga/definition.lua:69: attempt to concatenate local 'short_name' (a nil value)
stack traceback:
        .../pack/packer/opt/lspsaga.nvim/lua/lspsaga/definition.lua:69: in function 'handler'
        /usr/share/nvim/runtime/lua/vim/lsp.lua:1230: in function ''
        vim/_editor.lua: in function <vim/_editor.lua:0>
```

my config
```lua
        use { "glepnir/lspsaga.nvim",
            branch = "version_2",
            after = 'nvim-lspconfig',
            config = [[ require "LSP.lsp-saga" ]]
        }
```

Do I need to give more info?

**Comment by NEX-S** [comment]
```lua
require "lspsaga".init_lsp_saga {
    border_style = "single",
    move_in_saga = { next = '<C-n>', prev = '<C-p>' },
    diagnostic_header = { " ", " ", " ", " " },
    show_diagnostic_source = true,
    -- add bracket or something with diagnostic source, just have 2 elements
    diagnostic_source_bracket = {},
    -- use emoji lightbulb in default
    code_action_icon = "",
    -- if true can press number to execute the codeaction in codeaction window
    code_action_num_shortcut = true,
    -- same as nvim-lightbulb but async
    code_action_lightbulb = {
        enable = true,
        sign = true,
        sign_priority = 20,
        virtual_text = false,
    },
    finder_icons = {
        def = '  ',
        ref = '諭 ',
        link = '  ',
    },
    -- preview lines of lsp_finder and definition preview
    max_preview_lines = 16,
    finder_action_keys = {
        open = "o",
        vsplit = "s",
        split = "i",
        tabe = "t",
        quit = "q",
        scroll_down = "<C-f>",
        scroll_up = "<C-b>", -- quit can be a table
    },
    code_action_keys = {
        quit = "<ESC>",
        exec = "l",
    },
    rename_action_quit = "<C-c>",
    definition_preview_icon = "  ",
    symbol_in_winbar = {
        in_custom = false,
        enable = false,
        separator = ' ',
        show_file = true,
        click_support = false,
    },
    saga_winblend = 20,
}


local opt = { noremap = true, silent = true }
vim.keymap.set('n', "ga",    require "lspsaga.codeaction".code_action,        opt )
vim.keymap.set('n', "gh",    require "lspsaga.signaturehelp".signature_help,  opt )
vim.keymap.set('n', "gD",    require "lspsaga.finder".lsp_finder,             opt )
vim.keymap.set('n', "go",    require "lspsaga.hover".render_hover_doc,        opt )
vim.keymap.set('n', "<C-r>", require "lspsaga.rename".lsp_rename,             opt )
vim.keymap.set('n', "gd",    require "lspsaga.definition".preview_definition, opt )

vim.keymap.set('n', ";d",    require "lspsaga.diagnostic".show_line_diagnostics, opt )
vim.keymap.set('n', "<A-p>", require "lspsaga.diagnostic".goto_prev, opt )
vim.keymap.set('n', "<A-n>",  require "lspsaga.diagnostic".goto_next, opt )

vim.keymap.set("n", ";x", "<CMD>Lspsaga open_floaterm<CR>", opt )
vim.keymap.set("t", "<ESC>", "<CMD>Lspsaga close_floaterm<CR>", opt )
```

**Comment by glepnir** [comment]
@NEX-S can you give the full file path ? you can print it in neovim `echo expand('%:p')`

**Comment by NEX-S** [comment]
`/home/nex/.config/nvim/init.lua` is my nvim config entrance.
`/home/nex/.config/nvim/lua/LSP/lsp-saga.lua` is my saga config
`/home/nex/.config/nvim/lua/core/plugins.lua` is my plugins config
`/home/nex/Git/C/test.c` is my test.c file

**Comment by glepnir** [comment]
now should be fixed. version2 is develop branch . when it's complete. I will merge to main.

**Comment by NEX-S** [comment]
It's working now, but i will got this message before floating window popup.
```
Error executing vim.schedule lua callback: .../pack/packer/opt/lspsaga.nvim/lua/lspsaga/definition.lua:91: attempt to index upvalue 'self' (a nil value)
stack traceback:
        .../pack/packer/opt/lspsaga.nvim/lua/lspsaga/definition.lua:91: in function 'handler'
        /usr/share/nvim/runtime/lua/vim/lsp.lua:1230: in function ''
        vim/_editor.lua: in function <vim/_editor.lua:0>
```

you can checkout my nvim-config repo to get my full config :D

**Comment by glepnir** [comment]
@NEX-S  Please use command not call function  there has some breakchagne of function call. 

```lua
vim.keymap.set('n', "gd",    require "lspsaga.definition".preview_definition, opt )
-- use command in rhs
Lspsaga preview_definition
```
also you need change other rhs of your kemap

**Comment by NEX-S** [comment]
okay, i've found your bilibili stream room, 感谢大佬对nvim社区的贡献，关注力 <3

**Comment by NEX-S** [comment]
i've change my config to `vim.keymap.set('n', "gd", "<CMD>Lspsaga preview_definition<CR>", opt )` but got this...
```
Error executing vim.schedule lua callback: .../pack/packer/opt/lspsaga.nvim/lua/lspsaga/definition.lua:29: attempt to index a nil value
stack traceback:
        .../pack/packer/opt/lspsaga.nvim/lua/lspsaga/definition.lua:29: in function 'handler'
        /usr/share/nvim/runtime/lua/vim/lsp.lua:1230: in function ''
        vim/_editor.lua: in function <vim/_editor.lua:0>
Press ENTER or type command to continue
```

**Comment by kohane27** [comment]
I still have this problem after updating to [fix #446 · glepnir/lspsaga.nvim@eec313a](https://github.com/glepnir/lspsaga.nvim/commit/eec313a3e8be75b2727619368e1e4bd46b1c13c9)

Nothing happens when I issue the command `Lspsaga preview_definition`. No error but also no window shown.

```
❯ nvim --version
NVIM v0.7.2
Build type: Release
LuaJIT 2.1.0-beta3
Compiled by builduser
```

- Operating system: Linux

Any input is much appreciated. Thank you!

**Comment by glepnir** [comment]
@kohane27  you need provide more information and I don't think your problem same as this issue
